### PR TITLE
fixed: cube server's availability zone(and other resources that are u…

### PR DESCRIPTION
**Pull request by gabisavu98**
…sing availability zone) empty string check.

## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/resource_server.go [hunk 1]
```diff
@@ -50,8 +50,10 @@ func resourceServer() *schema.Resource {
 				Computed: true,
 			},
 			"availability_zone": {
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.All(validation.StringInSlice([]string{"AUTO", "ZONE_1", "ZONE_2", "ZONE_3"}, true)),
```
Comment (mihaela-mh) [inline ionoscloud/resource_server.go:56]: From what I see in the swagger file, for **ServerProperties** we have only _AUTO, ZONE_1, ZONE_2,_ and not _ZONE_3._ Not sure if ZONE_3 works on the API, but we could address this inconsistency and update here or on the swagger file.

File: ionoscloud/resource_k8s_node_pool.go [hunk 2]
```diff
@@ -164,7 +164,7 @@ func resourceK8sNodePool() *schema.Resource {
 				Description:  "The compute availability zone in which the nodes should exist",
 				Required:     true,
 				ForceNew:     true,
-				ValidateFunc: validation.All(validation.StringIsNotWhiteSpace),
+				ValidateFunc: validation.All(validation.StringInSlice([]string{"AUTO", "ZONE_1", "ZONE_2", "ZONE_3"}, true)),
```
Comment (mihaela-mh) [inline ionoscloud/resource_k8s_node_pool.go:167]: Same as the comment from resource_server.go

File: ionoscloud/resource_volume.go [hunk 3]
```diff
@@ -78,9 +78,10 @@ func resourceVolume() *schema.Resource {
 				Optional: true,
 			},
 			"availability_zone": {
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.All(validation.StringInSlice([]string{"AUTO", "ZONE_1", "ZONE_2", "ZONE_3"}, true)),
```
Comment (mihaela-mh) [inline ionoscloud/resource_volume.go:84]: Only on VolumeProperties we have **ZONE_3**, so this one should be alright ✔️

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
-## 6.2.2
+## 6.2.2(Upcoming Release)
```
Comment (IuliaBlindu) [inline CHANGELOG.md:1]: 6.2.2 is already released, now it's 6.2.3

**Review by IuliaBlindu — APPROVED** [review]

**Review by cristiGuranIonos — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (gabisavu98) [reply CHANGELOG.md]: I changed it

File: ionoscloud/resource_server.go (hunk 1 above)
Comment (gabisavu98) [reply ionoscloud/resource_server.go:56]: I checker now and ZONE_3 isn't

**Pull request merged**